### PR TITLE
Salesloft Structured Data Records

### PR DESCRIPTION
**Pull request by amcpanaligan**
For #28 

@iamEAP Tried a different approach to make the `custom_fields` formatting a little bit easier. Downloaded the `flat` plugin that flattens the supplied nested object. The custom fields appear as `custom_field.field` and so on.

Let me know if this works. This will eliminate the potential complexity of writing it as a record response

### CODE REVIEW

File: src/steps/person/person-delete.ts [hunk 1]
```diff
@@ -15,6 +15,17 @@ export class DeletePersonStep extends BaseStep implements StepInterface {
     description: 'Person\'s email address',
   }];
 
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'person',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
```
Comment (iamEAP) [inline src/steps/person/person-delete.ts]: - [ ] Also recommend adding `created_at` and `updated_at`

File: src/steps/person/person-delete.ts [hunk 2]
```diff
@@ -15,6 +15,17 @@ export class DeletePersonStep extends BaseStep implements StepInterface {
...
+      field: 'id',
+      type: FieldDefinition.Type.STRING,
```
Comment (iamEAP) [inline src/steps/person/person-delete.ts]: - [ ] The id field appears to be `NUMERIC`.

File: src/steps/person/person-create-or-update.ts [hunk 3]
```diff
@@ -14,6 +14,16 @@ export class CreateOrUpdatePersonStep extends BaseStep implements StepInterface
     type: FieldDefinition.Type.MAP,
     description: 'A map of field names to field values',
   }];
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'person',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
+      field: 'id',
+      type: FieldDefinition.Type.STRING,
```
Comment (iamEAP) [inline src/steps/person/person-create-or-update.ts]: - [ ] `NUMERIC` type

File: src/steps/person/person-delete.ts (hunk 1 above)
Comment (amcpanaligan) [reply src/steps/person/person-delete.ts]: @iamEAP Thanks for your feedback. We can flatten the `custom_fields` easily for now.

**Pull request merged**